### PR TITLE
added react and one chart component

### PR DESCRIPTION
**Pull request by gabrielkay**
- Added React with Webpacker
- Created a MonthlyProgressChart component
- Added a charts.js pack
- Added labels to the MonthlyProgressChart
- Renders the component in goals/show
Paired with @nhunzaker for this

### CODE REVIEW

File: app/javascript/packs/application.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+
+// Uncomment to copy all static images under ../images to the output folder and reference
+// them with the image_pack_tag helper in views (e.g <%= image_pack_tag 'rails.png' %>)
+// or the `imagePath` JavaScript helper below.
+//
+// const images = require.context('../images', true)
+// const imagePath = (name) => images(name, true)
+
+console.log('Hello World from Webpacker')
```
Comment (nhunzaker) [inline app/javascript/packs/application.js]: You could probably delete this file, if you wanted.

File: app/javascript/packs/hello_react.jsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+Hello.propTypes = {
+  name: PropTypes.string
+}
+
+document.addEventListener('DOMContentLoaded', () => {
+  ReactDOM.render(
+    <Hello name="React" />,
+    document.body.appendChild(document.createElement('div')),
+  )
+})
```
Comment (nhunzaker) [inline app/javascript/packs/hello_react.jsx]: Likewise with this module. What about deleting it?

File: package.json [hunk 3]
```diff
@@ -1,5 +1,16 @@
 {
   "name": "daily",
   "private": true,
-  "dependencies": {}
+  "dependencies": {
+    "@babel/preset-react": "^7.0.0",
+    "@rails/webpacker": "^4.0.7",
+    "babel-plugin-transform-react-remove-prop-types": "^0.4.24",
+    "prop-types": "^15.7.2",
```
Comment (nhunzaker) [inline package.json]: These dependencies could probably go too (`prop-types` and `babel-plugin-transform-react-remove-prop-types`), but if you do that, be sure to remove the entry in the babel config here:

https://github.com/gabrielkay/daily/pull/18/files#diff-fc10a6682269c1fefc2abe05d1500b8aR77

File: app/javascript/packs/hello_react.jsx (hunk 2 above)
Comment (gabrielkay) [reply app/javascript/packs/hello_react.jsx]: Just removed the unnecessary files and dependancies, and can now reference this PR if I want to check the info in the auto-generated hello_world files 👍

**Pull request merged**